src/wallet.js

```diff
@@ -630,13 +651,17 @@ const wallet = {
    * @inner
    */
   changePin(oldPin, newPin) {
+    const accessData = this.getWalletAccessData();
+
+    if (accessData['from_xpub']) {
```

Comment:
Create a method `isFromXpub()` and use it in all these verifications.

Reply:
Done